### PR TITLE
Implement GiftCardAdjustment API

### PR DESCRIPTION
**Pull request by sharoonthomas**
Implements the GiftCardAdjustment API with Tests.

* Adds an `adjust` method on gift card
* Also implements a `search` method on gift card for easy lookup of gift cards

### CODE REVIEW

**Comment by sharoonthomas** [comment]
@jamiemtdwyer requesting review.

File: shopify/resources/gift_card.py [hunk 1]
```diff
@@ -5,3 +5,48 @@ class GiftCard(ShopifyResource):
...
+           query: Text to search for gift cards
+           page: Page to show (default: 1)
+           limit: Amount of results (default: 50) (maximum: 250)
+           fields: comma-seperated list of fields to include in the response
+        Returns:
+           An array of gift cards.
+        """
+        return cls._build_list(cls.get("search", **kwargs))
+
+    def adjust(self, amount, **kwargs):
```
Comment (jamiemtdwyer) [inline shopify/resources/gift_card.py]: I'd prefer to see this be `add_adjustment` for semantical reasons

File: shopify/resources/gift_card.py [hunk 2]
```diff
@@ -5,3 +5,48 @@ class GiftCard(ShopifyResource):
...
+                  that can be attached to the adjustment.
+                  This value is not visible to customers.
+            remote_transaction_url: An optional reference URL that can be
+                  attached to the adjustment.
+                  This value is not visible to customers.
+        """
+        data = kwargs
+        data['amount'] = amount
+        data['gift_card_id'] = self.id
+        return Adjustments.create(data)
```
Comment (jamiemtdwyer) [inline shopify/resources/gift_card.py]: would it be possible to refactor this to be more similar to https://github.com/Shopify/shopify_python_api/blob/master/shopify/resources/price_rule.py#L5-L7

File: shopify/resources/gift_card.py (hunk 1 above)
Comment (sharoonthomas) [reply shopify/resources/gift_card.py]: 🆗

**Comment by sharoonthomas** [comment]
@jamiemtdwyer made the changes you asked for. Thanks for the super quick review.

**Comment by jamiemtdwyer** [comment]
Great, I will merge this in tomorrow and release in the newest version.

Thanks for making the suggested changes @sharoonthomas!

**Comment by sharoonthomas** [comment]
Just a polite reminder if this fell off your radar.

**Comment by jamiemtdwyer** [comment]
Hey @sharoonthomas, sorry about the delay. Thanks for your contribution!

**Pull request merged**